File: leetcode/1828.统计一个圆中点的数目.py
#
# @lc app=leetcode.cn id=1828 lang=python3
#
# [1828] 统计一个圆中点的数目
#
from typing import List


# @lc code=start
class Solution:
    def countPoints(self, points: List[List[int]], queries: List[List[int]]) -> List[int]:
        res = []
        for xj, yj, rj in queries:
            cur_count = 0
            for xi, yi in points:
                if (xj - xi) ** 2 + (yj - yi) ** 2 > rj ** 2:
                    continue
                cur_count += 1
            res.append(cur_count)
        return res


# @lc code=end
print(Solution().countPoints(
    points=[[1, 3], [3, 3], [5, 3], [2, 2]],
    queries=[[2, 3, 1], [4, 3, 1], [1, 1, 2]]
))

File: leetcode/1828.统计一个圆中点的数目.cpp
#include <iostream>
#include <vector>
using namespace std;
/*
 * @lc app=leetcode.cn id=1828 lang=cpp
 *
 * [1828] 统计一个圆中点的数目
 */

// @lc code=start
class Solution
{
public:
    vector<int> countPoints(vector<vector<int>> &points, vector<vector<int>> &queries)
    {
        vector<int> res;
        for (auto &&query : queries)
        {
            int cur_count = 0;
            for (auto &&point : points)
            {
                if ((point[0] - query[0]) * (point[0] - query[0]) + (point[1] - query[1]) * (point[1] - query[1]) <= query[2] * query[2])
                {
                    cur_count++;
                }
            }
            res.push_back(cur_count);
        }
        return res;
    }
};
// @lc code=end
